pkg/queue/instance.go

```diff
@@ -45,10 +50,12 @@ func NewQueue(errorDelay time.Duration) Instance {
 		delay:   errorDelay,
 		tasks:   make([]Task, 0),
 		closing: false,
+		closed:  true,
 		cond:    sync.NewCond(&sync.Mutex{}),
 	}
 }
 
+// Push enqueues a task
 func (q *queueImpl) Push(item Task) {
```

Comment:
I am worried about this scenario:
* lose leader
* 100k events are added to the queue, but we are not processing them. This leads to memory leak
* Become the leader - process all 100k events again, causing massive CPU spike *and* pushing bad events

Reply:
if leader lost, the queue is closed too.And then no task can be pushed in to the queue closed.